js/shopping-list.js

```diff
@@ -0,0 +1,50 @@
+var ulElement = document.querySelector('ul');
+
+function addListItems(event){
+    var form = event.target;
+    var items = form.querySelector('#items').value.split(',');
```

Comment:
We could refactor this code to the following:
![Capture](https://user-images.githubusercontent.com/20998142/84095775-0538ba80-a9b5-11ea-8e34-740ad00969d0.PNG)

[The Array.prototype.map method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) declares a new Array, iterates over each element of the Array you called `.map` on, then pushes whatever your function returns for that given item to the new Array, then returns the new Array. In short, it passes each element of the original Array into your provided function, then transforms it to whatever your function returns.

[String.prototype.trim](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim) removes whitespace from the beginning and end of a String. So what we're ultimately saying is, "Take this String, split it on commas, then for each item, trim all the whitespace from the start and end."